File: AdventOfCodeIn2019InPython/AdventOfCode20191201_1.py
def main():
    mass_list = read_input("Advent20191201_1_input.txt")
    fuel_list = [required_fuel(x) for x in mass_list]
    total_fuel = sum(fuel_list)
    print(total_fuel)

def read_input(filename):
    with open(filename, "r") as file:
        contents = [line for line in file]
    return [int(x) for x in contents]

def required_fuel(mass):
    return (mass // 3) - 2

if __name__ == "__main__":
    main()

File: AdventOfCode2019InHaskell/src/AdventOfCode20191201_1.hs
module AdventOfCode20191201_1
    ( simpleFuelCaculation
    ) where
        
import System.IO

simpleFuelCaculation :: IO ()
simpleFuelCaculation = do
    inputText <- readFile "Advent20191201_1_input.txt" 
    requiredFuel <- return (totalRequiredFuel inputText)
    putStrLn (show requiredFuel)

totalRequiredFuel :: String -> Int
totalRequiredFuel = sum . fuelList . massList

massList :: String -> [Int]
massList = (map read) . lines

fuelList :: [Int] -> [Int]
fuelList = map requiredFuel

requiredFuel :: Int -> Int
requiredFuel mass = max 0 ((quot mass 3) - 2)